### ISSUE TITLE
Do Not Disturb ignores Chromium-based browser notifications

### ISSUE THREAD

**Issue by dodalovicgran** [pop-os/gnome-shell#133]
Hi, there's this setting:
![image](https://user-images.githubusercontent.com/116893230/232154683-6c687853-0807-4f6f-bcab-e8d0f21cdb85.png)

Although it's turned on, I'm still getting notifications. This confuses me. 

Thank You

**Comment by jacobgkau** [comment]
This is not related to Pop Shell (the window tiling extension), so I'm moving it to the general Pop!_OS repository.

The setting tells GNOME Shell not to pop up notifications in the top-center of the screen (like "silent mode" for your phone.)

It can be toggled through GNOME Control Center (the Settings app) as shown in your screenshot, or at the bottom-left of the calendar/notifications drop-down that's part of GNOME Shell at the top-center of the screen.

**Comment by dodalovicgran** [comment]
So, is it designed to be broken or it's not a question for the POP team?

**Comment by leviport** [comment]
>So, is it designed to be broken

No, it is not "designed to be broken". The kinds of notifications you still receive will be helpful. It suppresses `notify-send` notifications on my end, so whichever notifications you're seeing might be a different type, coming from a different application.

**Comment by dodalovic** [comment]
Ok
It's far from ideal that user is unable to decide that under some circumstances it's absolutely not acceptable to get anything shown.

Like, I have to hold a webinar and I'm most likely to get some rubbish shown on the screen, 

There needs to be a Don't disturb ON/OFF toggle, and additionally when it's on there's a sub toggle to allow only some apps. I agree that 99% of the time the way it works now is great. The issue is that me being unable to have a simple switch to totally forbid anything shown is making me sad

**Comment by leviport** [comment]
Yeah, I understand that. I'm asking what applications are sending notifications that are not blocked by DnD mode. Like I said, `send-notify`, which is a way to test notifications in Gnome, does not display notifications on my machine while DnD mode is active. If you can tell us the names of applications that are _not_ blocked by DnD mode, it would be extremely helpful. If we can confirm that behavior, we can refine and reopen this issue.

**Comment by dodalovicgran** [comment]
I have no idea? Browser?

**Comment by leviport** [comment]
In order for us to understand the nature of the bug, we'll have to first be able to understand the nature of the notifications. If you have more info later, please share it with us so that we can investigate effectively.

**Comment by dodalovicgran** [comment]
What is the "nature of the notifications"? I believe that Google Calendar sends me something, Reddit comment or what not...

**Comment by leviport** [comment]
We need exact details. What browser? What site? What happened that generated the notification?

**Comment by dodalovicgran** [comment]
Chrome or Edge
It comes from Google Calendar I think. 
Or even if someone comments on a Reddit post or similar.

**Comment by jacobgkau** [comment]
There is an upstream issue in GNOME regarding Chrome/Chromium-based browser notifications: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/2148 

We can reopen this issue for tracking purposes and rename it to reflect the issue you were trying to report, but the issue will probably need to be fixed upstream.

**Comment by jacobgkau** [comment]
Also transferring to our GNOME Shell repo since that is where the upstream issue is located.

Reading more into the upstream issue page I linked, it sounds like the GNOME developers consider this to be a bug in Chromium, since Chromium is marking the notifications as critical (which does sound like incorrect behavior to me.)

There is also a Chromium bug report: https://bugs.chromium.org/p/chromium/issues/detail?id=1093556 

But Chromium marked it as `WontFix` and said it's up to the individual websites to flag the notification priorities properly.

**Comment by jacobgkau** [comment]
My instinct would be that the web browser would be the most logical place to filter for incorrect notification priorities from websites, since the web is much less curated than apps are and it's much easier to browse an incorrectly-coded website than it is to install an incorrectly-coded app. However, I also think it would be good for the DE to allow overriding/ignoring priorities for specific apps at the user's discretion.

In the meantime, I assume you should be able to turn desktop notifications off in Chromium for the websites that use critical priority? (E.g. I don't ever get notifications from Reddit, Google Calendar, etc. in a Chromium-based browser because I deny notification permissions and just use the on-page notification interfaces instead.) You should be able to do this per-site using one of the buttons in the URL bar, or Chrome has global settings where you can also specify specific websites: https://support.google.com/chrome/answer/3220216
